### PR TITLE
Add SetTags and RemoveTag to the ITaggable interface

### PR DESCRIPTION
**Pull request by HarveyLink**
ADO work item: https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/5176

### CODE REVIEW

File: Azure.ResourceManager.Core/GenericResourceData.cs [hunk 1]
```diff
@@ -74,5 +77,25 @@ public GenericResourceData(ResourceIdentifier id, LocationData location)
...
+            other.Model.Tags.Clear();
+            foreach (var tag in other.Tags)
+            {
+                other.Model.Tags.Add(tag);
+            }
+
+            return other.Model;
+        }
     }
 }
```
Comment (m-nash) [inline Azure.ResourceManager.Core/GenericResourceData.cs:101]: missing newline

File: Azure.ResourceManager.Core/GenericResourceOperations.cs [hunk 2]
```diff
@@ -141,5 +142,53 @@ public override async Task<ArmResponse<GenericResource>> GetAsync(CancellationTo
         public override void Validate(ResourceIdentifier identifier)
         {
         }
+
+        /// <inheritdoc/>
+        public ArmResponse<GenericResource> SetTags(IDictionary<string, string> tags)
+        {
+            throw new System.NotImplementedException();
```
Comment (m-nash) [inline Azure.ResourceManager.Core/GenericResourceOperations.cs:149]: any reason these are stil not implemented?

File: Azure.ResourceManager.Core/ITaggableResource.cs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System.Collections.Generic;
```
Comment (m-nash) [inline Azure.ResourceManager.Core/ITaggableResource.cs:4]: should add the non LRO versions of add to
AddTag
AddTagAsync

File: Azure.ResourceManager.Core/ResourceGroupOperations.cs [hunk 4]
```diff
@@ -230,6 +230,54 @@ public async Task<ArmOperation<ResourceGroup>> StartAddTagAsync(string name, str
             return container.CreateAsync(name, model, cancellationToken);
         }
 
+        /// <inheritdoc/>
+        public ArmResponse<ResourceGroup> SetTags(IDictionary<string, string> tags)
+        {
+            throw new NotImplementedException();
```
Comment (m-nash) [inline Azure.ResourceManager.Core/ResourceGroupOperations.cs:236]: should not be empty implementations

File: azure-proto-compute/AvailabilitySetOperations.cs [hunk 5]
```diff
@@ -180,6 +180,46 @@ public Task<ArmOperation<AvailabilitySet>> StartAddTagAsync(string key, string v
             return UpdateAsync(patchable);
         }
 
+        public ArmResponse<AvailabilitySet> SetTags(IDictionary<string, string> tags)
+        {
+            throw new System.NotImplementedException();
```
Comment (m-nash) [inline azure-proto-compute/AvailabilitySetOperations.cs:185]: should not be empty implementations

File: azure-proto-compute/VirtualMachineOperations.cs [hunk 6]
```diff
@@ -287,6 +287,46 @@ public async Task<ArmOperation<VirtualMachine>> StartAddTagAsync(string key, str
                 v => new VirtualMachine(this, new VirtualMachineData(v)));
         }
 
+        public ArmResponse<VirtualMachine> SetTags(IDictionary<string, string> tags)
+        {
+            throw new System.NotImplementedException();
```
Comment (m-nash) [inline azure-proto-compute/VirtualMachineOperations.cs:292]: should not be empty implementations

File: azure-proto-network/NetworkInterfaceOperations.cs [hunk 7]
```diff
@@ -153,6 +153,46 @@ public async Task<ArmOperation<NetworkInterface>> StartAddTagAsync(string key, s
                 n => new NetworkInterface(this, new NetworkInterfaceData(n)));
         }
 
+        public ArmResponse<NetworkInterface> SetTags(IDictionary<string, string> tags)
+        {
+            throw new System.NotImplementedException();
```
Comment (m-nash) [inline azure-proto-network/NetworkInterfaceOperations.cs:158]: should not be empty implementations

File: azure-proto-network/NetworkSecurityGroupOperations.cs [hunk 8]
```diff
@@ -158,6 +158,51 @@ public async Task<ArmOperation<Response>> StartDeleteAsync(CancellationToken can
...
         }
 
+        private protected virtual NetworkSecurityGroup GetResource()
+        {
+            return Get().Value;
+        }
+
+        public ArmResponse<NetworkSecurityGroup> SetTags(IDictionary<string, string> tags)
+        {
+            throw new System.NotImplementedException();
```
Comment (m-nash) [inline azure-proto-network/NetworkSecurityGroupOperations.cs:168]: should not be empty implementations

File: azure-proto-network/PublicIpAddressOperations.cs [hunk 9]
```diff
@@ -151,6 +151,46 @@ public async Task<ArmOperation<PublicIpAddress>> StartAddTagAsync(string key, st
                 n => new PublicIpAddress(this, new PublicIPAddressData(n)));
         }
 
+        public ArmResponse<PublicIpAddress> SetTags(IDictionary<string, string> tags)
+        {
+            throw new System.NotImplementedException();
```
Comment (m-nash) [inline azure-proto-network/PublicIpAddressOperations.cs:156]: should not be empty implementations

File: azure-proto-network/VirtualNetworkOperations.cs [hunk 10]
```diff
@@ -175,6 +175,46 @@ public SubnetContainer GetSubnetContainer()
             return new SubnetContainer(this);
         }
 
+        public ArmResponse<VirtualNetwork> SetTags(IDictionary<string, string> tags)
+        {
+            throw new System.NotImplementedException();
```
Comment (m-nash) [inline azure-proto-network/VirtualNetworkOperations.cs:180]: should not be empty implementations

**Review by m-nash — CHANGES_REQUESTED** [review]

File: Azure.ResourceManager.Core/ITaggableResource.cs (hunk 3 above)
Comment (HarveyLink) [reply Azure.ResourceManager.Core/ITaggableResource.cs:4]: Will open another PR for work item: https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/5177

**Review by m-nash — APPROVED** [review]

**Pull request merged**